### PR TITLE
fix(radio): can't select radio on click

### LINKED ISSUE #9136: igx-radio emits multiple change event with ngModel
## Description  
Using igx-radio and ngModel does not work as expected. Inside the method triggered by event "change" we set up the value and trigger again the event. In version 10.1.4 wasn't doing that.

 * igniteui-angular version: 10.2.15
 * browser: Microsoft Edge Version 89.0.774.50

## Steps to reproduce  

1. Navigate to stackblitz example: https://stackblitz.com/edit/angular-igniteui-radio?file=src%2Fapp%2Fapp.module.ts
2. Select Option 1
3. check logs, select again option 2

## Result  
Changing the ngModel does not trigger the change event

## Expected result  
Changing the ngModel does not trigger the change event

## Attachments  
1. Navigate to stackblitz example:
https://stackblitz.com/edit/angular-igniteui-radio?file=src%2Fapp%2Fapp.module.ts

### PR DESCRIPTION
**Pull request by simeonoff**
Closes #9136  

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/radio/radio.component.html [hunk 1]
```diff
@@ -9,10 +9,10 @@
     [attr.aria-checked]="checked"
     [attr.aria-labelledby]="ariaLabelledBy"
     [attr.aria-label]="ariaLabel"
-    (click)="_clicked($event)"
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/radio/radio.component.html:12]: Shouldn't the input itself still be handling click together with the label?

Comment (simeonoff) [reply projects/igniteui-angular/src/lib/radio/radio.component.html:12]: The native input is hidden, hence why it doesn't make sense to listen for the click even on it.

**Review by kdinev — DISMISSED** [review]

**Review by kdinev — DISMISSED** [review]

**Review by wnvko — CHANGES_REQUESTED** [review]
`igxRadio` now calls `select` method on each click. This forces `ngModelChange` to fire even when value was not changed which is not correct. Also changing the value of the bound property does not selects the radio.  Having this radios:
```
        <igx-radio [(ngModel)]="selected" value="option1">Option 1</igx-radio>
        <igx-radio [(ngModel)]="selected" value="option2">Option 2</igx-radio>
        <button (click)="selectSecond()">Select second</button>
```
and this handler
```
    public selectSecond() {
        this.selected = 'Option 2';
    }
```
does not selects second option.

**Comment by simeonoff** [comment]
> `igxRadio` now calls `select` method on each click. This forces `ngModelChange` to fire even when value was not changed which is not correct.

The radio has always called `select` on each click. This is nothing new - no behavior change with this PR from that perspective. What's being changed now is that `select` is also called on `keyUp` to account for changing the radios using keyboard as if they were in a group.

>Also changing the value of the bound property does not selects the radio. Having this radios:
> 
> ```
>         <igx-radio [(ngModel)]="selected" value="option1">Option 1</igx-radio>
>         <igx-radio [(ngModel)]="selected" value="option2">Option 2</igx-radio>
>         <button (click)="selectSecond()">Select second</button>
> ```
> 
> and this handler
> 
> ```
>     public selectSecond() {
>         this.selected = 'Option 2';
>     }
> ```
> 
> does not selects second option.

It should be `this.selected = 'option2'` instead. The value is `option2` not `Option 2`.

**Review by wnvko — DISMISSED** [review]

**Review by wnvko — APPROVED** [review]

**Pull request merged**